docs/shared/angular-standalone-tutorial/angular-standalone.md

```diff
@@ -0,0 +1,795 @@
+- you could go even further and configure routes within the libraries and only import and attach those routes to the application routing mechanism.
+
+## Visualizing your Project Structure
+
+Nx automatically detects the dependencies between the various parts of your workspace and builds a [project graph](/core-features/explore-graph). This graph is used by Nx to perform various optimizations such as determining the correct order of execution when running tasks like `nx build`, identifying [affected projects](/core-features/run-tasks#run-tasks-affected-by-a-pr) and more. Interestingly you can also visualize it.
+
+Just run:
+
+```shell
+nx graph
+```
+
+You should be able to see something similar to the following in your browser (hint: click the "Show all projects" button).
+
+{% graph height="450px" %}
+
+```json
+{
+  "projects": [
+    {
+      "name": "myngapp",
+      "type": "app",
+      "data": {
+        "tags": []
+      }
+    },
+    {
+      "name": "e2e",
+      "type": "e2e",
+      "data": {
+        "tags": []
+      }
+    },
+    {
+      "name": "modules-shared-ui",
+      "type": "lib",
+      "data": {
+        "tags": []
+      }
+    },
+    {
+      "name": "modules-orders",
+      "type": "lib",
+      "data": {
+        "tags": []
+      }
+    },
+
+    {
+      "name": "modules-products",
+      "type": "lib",
+      "data": {
+        "tags": []
+      }
+    }
+  ],
+  "dependencies": {
+    "myngapp": [
+      { "source": "myngapp", "target": "modules-orders", "type": "dynamic" },
+      { "source": "myngapp", "target": "modules-products", "type": "dynamic" }
+    ],
+    "e2e": [{ "source": "e2e", "target": "myngapp", "type": "implicit" }],
+    "modules-shared-ui": [],
+    "modules-orders": [],
+    "modules-products": []
+  },
+  "workspaceLayout": { "appsDir": "", "libsDir": "" },
+  "affectedProjectIds": [],
+  "focus": null,
+  "groupByFolder": false
+}
+```
+
+{% /graph %}
+
+Notice how `modules-shared-ui` is not yet connected to anything because we didn't import it in any of our projects. Also the arrows to `modules-orders` and `modules-products` are dashed because we're using lazy imports.
+
+Exercise for you: change the codebase such that `modules-shared-ui` would be connected from `modules-orders` and `modules-products`. Note: you need to restart the `nx graph` command to refresh or run it with the `--watch` flag.
+
+## Imposing Constraints with Module Boundary Rules
+
+Once you modularize your codebase you want to make sure that the modules are not coupled to each other in an uncontrolled way. Here are some examples of how we might want to guard our small demo workspace:
+
+- we might want to allow `modules-orders` to import from `modules-shared-ui` but not the other way around
+- we might want to allow `modules-orders` to import from `modules-products` but not the other way around
+- we might want to allow all libraries import the `modules-shared-ui` components, but not the other way around
```

Comment:
Yeah I see what you mean 🤔 . Here I mostly just wanted to lay out the conditions regardless of the technical implementation.

Further down when we add the rules, there are some differences that show how a feature-library which is more end-to-end hence allows multiple dependencies, while a `type:ui` is a library at the edges of the project tree, thus having a much more restricted allowance for libs it can depend on.